File: C#/Lv_0/배열의 원소만큼 추가하기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] arr) {
        List<int> answer = new List<int>();
        foreach(int i in arr) for(int j=0; j<i; j++) answer.Add(i);
        return answer.ToArray();
    }
}

// Linq를 활용한 더 좋은 방법
// public class Solution {
//     public int[] solution(int[] arr) 
//     {
//         List<int> answer = new List<int>();
//         foreach(int n in arr)
//             answer.AddRange(Enumerable.Repeat(n, n));
//         return answer.ToArray();
//     }
// }

File: JavaScript/Lv_0/배열의 원소만큼 추가하기.js
function solution(arr) {
    var answer = [];
    for(let i=0; i<arr.length; i++)
        for(let j=0; j<arr[i]; j++)
            answer.push(arr[i]);
    return answer;
}

// 더 간단한 방법, fill을 사용
// function solution(arr) {
//     return arr.reduce((list, num) => [...list, ...new Array(num).fill(num)], []);
// }